crates/transaction-pool/src/maintain.rs

```diff
@@ -584,8 +584,13 @@ where
 
     let pool_transactions = txs_signed
         .into_iter()
-        .filter_map(|tx| tx.try_ecrecovered().map(<P::Transaction>::from_recovered_transaction))
+        .filter_map(|tx| tx.try_ecrecovered())
+        .filter_map(|tx| {
+            // Filter out errors
+            <P as TransactionPool>::Transaction::try_from_recovered_transaction(tx).ok()
+        })
```

Comment:
I'm a bit unsure how to handle this. Is it ok to filter errors here?

Reply:
this is okay here